cylc/flow/param_expand.py

```diff
@@ -261,7 +261,7 @@ class GraphExpander(object):
 
     _REMOVE = -32768
     _REMOVE_REC = re.compile(
-        r'(?:^|\s*=>).*' + str(_REMOVE) + r'.*?(?:$|=>\s*?)')
+        r'(?:^|\s*=>).*' + str(_REMOVE) + r'.*?(?:$|=>\s*?|&\s*)')
```

Comment:
Simpler fix. Just stop the greedy regex once it finds a) end of stream $, b) an arrow followed by spaces =>\s*, or c) ampersand symbol followed by spaces &\s*.

The last c) item covers the bug reported in the issue linked.

Reply:
Would these solutions handle the other forms of the problem e.g:

* `baz & foo<m-1> => foo<m>`
* `baz & foo<m-1> & pub => foo<m>`
* `bar & foo<m-1> & pub<m-1> & qux => foo<m>`

Reply:
@oliver-sanders not really good at parsing the graph expressions, but here's the output of the examples you provided.

For

```
params_map = {'m': ["cat", "dog"]}
templates = {'m': '_%(m)s'}
```

- `baz & foo<m-1> => foo<m>`
  * this is expanded to `{'baz & foo_cat => foo_dog', ' foo_cat'}`
- `baz & foo<m-1> & pub => foo<m>`
  * this is expanded to `{'baz & foo_cat & pub => foo_dog', 'pub => foo_cat'}`
- `bar & foo<m-1> & pub<m-1> & qux => foo<m>`
  * this is expanded to `{'bar & foo_cat & pub_cat & qux => foo_dog', 'qux => foo_cat'}`

These look OK, but can't really say I'm sure these are correct (@hjoliver do these look correct?)

Reply:
The expansion of the parameterised items seems correct to me but there is a bit of a discrepancy with the non-parameterised items.

E.g in example (2) we get `pub => foo_cat` but not `baz => foo_cat` similar to example (3) but different to example (1) where we just get `foo_cat`.

I'm not entirely sure how we should expand these things, here are my guesses:

* `baz & foo<m-1> => foo<m>`
  * `baz & foo_cat => foo_dog`
  * `[foo_cat]`
  * `(baz => foo_cat)`
* `baz & foo<m-1> & pub => foo<m>`
  * `baz & foo_cat & pub => foo_dog`
  * `[foo_cat]`
  * `(baz => foo_cat)`
  * `(pub => foo_cat)`
* `bar & foo<m-1> & pub<m-1> & qux => foo<m>`
  * `bar & foo_cat & pub_cat & qux => foo_dog`
  * `[foo_cat]`
  * `(bar => foo_cat)`
  * `(qux => foo_cat)`

Where:

* `compulsory => dependency` - needed for graphing
* `[optional => line]` - not needed as task is referenced elsewhere but not harmful
* `(pre-initial-type => dependency)` - not sure if we should have these or not, but if so they should be consistent